### PR TITLE
FEAT: HR user, create site allowance based on payroll start and end date

### PR DESCRIPTION
**Pull request by NoBoneZ**
## Is this a Feature, Chore or Bug?
- [x] Feature
- [] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
To add the feature that allows the additional salary for site allowance be created with a diffrent cycle, apart form the generic 24-23.

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.
added controllers, to the payroll entry doctype, that creates , cancel and delete the site allowance based on the events carried out on the payroll entry

## Is there a business logic within a doctype?
    - [x] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.
Additional salary
Payroll entry

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [] Existing Data
- [x] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [x] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [x] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [x] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/api/doc_events.py [hunk 1]
```diff
@@ -244,3 +244,67 @@ def update_training_event_data(doc, method):
 				'training': doc.event_name,
 			})
 			doc_esm.save()
+
+
+def after_insert_of_payroll_entry(doc, method):
+	if doc.employees:
```
Comment (mymi14s) [inline one_fm/api/doc_events.py:250]: Did you test this, when payroll_entry is created, employees is not created, employees is added later. This approach is wrong.

Comment (mymi14s) [reply one_fm/api/doc_events.py:250]: All you need do is use the payroll_entry start and end date, check HR and Payroll settings for payroll cycle and project within the start and end date.
Then fetch employees in the projects fetched, use the start and end date to get site allowance

File: one_fm/api/doc_events.py [hunk 2]
```diff
@@ -244,3 +244,67 @@ def update_training_event_data(doc, method):
...
+				employee_details = frappe.db.sql("""
+    							SELECT employee, COUNT(attendance_date) as count
+    							FROM `tabAttendance`
+    							WHERE site = %s AND attendance_date BETWEEN %s AND %s AND status = %s AND employee IN %s GROUP BY employee
+								""", (site.name, start_date, end_date, "Present", tup_of_employee), as_dict=True)
+
+				if employee_details:
+					for employee in employee_details:
+							#calculate Monthly_site_allowance with the rate of allowance per day.
+							Monthly_site_allowance =  round(int(site.allowance_amount)/no_of_days, 3)*int(employee["count"])
```
Comment (mymi14s) [inline one_fm/api/doc_events.py:286]: I dont know your number of days but it should be total days attended(Present for each employee, seek help from Jamsheer

**Comment by mymi14s** [comment]
Add Jamsheer as reviewer he understands the scenario better

File: one_fm/api/doc_events.py (hunk 1 above)
Comment (NoBoneZ) [reply one_fm/api/doc_events.py:250]: Oh, ok. When i tested it, i added employees on creation, i will make changes

**Comment by NoBoneZ** [comment]
> Add Jamsheer as reviewer he understands the scenario better

ok, i have done that

**Pull request closed without merging**